### ISSUE TITLE
Support for Keyboard-Only Navigation

### ISSUE THREAD

**Issue by darrenhunt2** [Stellarium/stellarium#2075]
I have an accessibility request to support keyboard-only navigation with no mouse required.

Tabbing to navigate does not follow the display order in most windows. Certain options in many windows cannot be accessed with only a keyboard. (I used Narrator to highlight the tab order clearly.) Navigating between settings tabs in windows like Sky and Viewing Options does not work. Currently, there is no apparent way to select/unselect checkboxes (spacebar won't work), so checkbox settings without a keyboard shortcut can't be changed. There are some shortcuts that use a mouse and cannot be bound to other keys.

Is improved keyboard navigation a possibility in the future?

**Comment by axd1967** [comment]
related: #1834 #1971

**Comment by gzotti** [comment]
We have started some work in this direction which was as it seems never of interest until a few weeks ago. One obstacle is the so-far visual indifference between buttons which are currently in focus and those that aren't. I have found a solution for that which finally allows some tabbing sequence, after re-working each and every GUI file. Other tasks are however estimate more urgent, so this is not to be expected to be complete for the next release.  

Whichever speed this may take, mouse-free navigation is not our aim. This pointing device is still of fundamental importance in the desktop domain.

**Comment by MReaseClark** [comment]
> This pointing device is still of fundamental importance in the desktop domain.
For users with the ability to use the device yes, however there are many with disabilities that cannot make use of a precision pointing device.

Without fixing this accessibility issue, a college in the US at least (one that is state/government operated) cannot legally offer this software for use in classrooms by students - just as an example. (Since being useable by keyboard alone is part of Section 508 functional requirements.)

While that may not be a concern for most software, one such as this seems heavily geared towards education, but sadly many will be unable to make use of it.

**Comment by gzotti** [comment]
Sorry for the US legal system. What are his/her alternatives now?  Please understand https://github.com/Stellarium/stellarium/wiki/FAQ#why-dont-you-implement. I think I can speak for the three of us stating we are no experts in software accessibility issues.  We are always open to pull requests with working solutions developed by people who know what to develop for the respective user group, but the three of us are busy in daytime occupation and do this on weekends instead of observing.

**Comment by 10110111** [comment]
> Without fixing this accessibility issue, a college <...> cannot legally offer this software for use in classrooms

Rotation of the scene can be achieved by using arrow keys. There are also quite a lot of other accelerator keys.

For the GUI, unfortunately, the standard <kbd>Tab</kbd> and <kbd>Space</kbd> manipulators are not very usable in the Stellarium. To click inside the GUI windows, you can try the Mouse Keys platform feature, which is natively present [in Windows](https://support.microsoft.com/en-us/windows/use-mouse-keys-to-move-the-mouse-pointer-9e0c72c8-b882-7918-8e7b-391fd62adf33) and [in the Xorg server](https://wiki.gentoo.org/wiki/Xorg/Using_the_numeric_keyboard_keys_as_mouse).

**Comment by MReaseClark** [comment]
> We are always open to pull requests with working solutions developed by people who know what to develop for the respective user group, but the three of us are busy in daytime occupation and do this on weekends instead of observing.

Completely understandable of course. As this is a libre application maintained by volunteers there is of course no obligation to try to implement such things. I just wanted to add another angle to the discussion for why this may be valuable. I also work for (though my words do not necessarily represent) one such college, which is currently reviewing student facing software, and this program came up.

> Mouse Keys

Mouse Keys is nice, and can potentially allow applications to meet Section 508 functional requirements for users with limited manipulation/reach and strength, but won't work for users without vision. A natural question might be "what value does this application have for a user without vision", and while I may have my own opinions on that it's not for me to say. Certainly there is a lot of textual info in this software, and it's useful in general for learning about satellites.
